### ISSUE TITLE
Icon filename configuration

### ISSUE THREAD

**Issue by ciampo** [arthurbergmz/webpack-pwa-manifest#36]
At the moment icon files are always named `icon_[size].[fingerprint].[ext]`, but there are certain icons that are expected to have the same filename (e.g. `/favicon.ico`, `apple-touch-icon.png`).

It would be good if the plugin allowed more freedom in defining `filename` patterns at the icon config level. It would also be good if, by specifying the filename pattern, we could include the file hash.

This is an example of what the configuration could look like for the `icons` object:

```json
"icons": [
  {
    "src": "src/images/favicon.png",
    "sizes": [16, 32],
    "filename": "favicon-[size].[hash].[ext]"
  },
  {
    "src": "src/images/favicon.png",
    "sizes": [180],
    "ios": true,
    "filename": "apple-touch-icon.[ext]"
  } 
]
```

Please let me know what do you think about these suggestions. I'd be also interested in helping you coding these features.

Thanks

**Comment by arthurbergmz** [comment]
Implemented in **v4.0.0**.

**Comment by Jaid** [comment]
Is this really implemented now, @arthurbergmz? I am using v4.1.1 and this is the relevant part of my `node_modules/webpack-pwa-manifest/index.d.ts`:

```typescript
declare namespace WebpackPwaManifest {
    interface Icon {
        src: string;
        size?: string | number;
        sizes?: number[];
        destination?: string;
        ios?: boolean | 'default';
    }
}
```

I can't find the property `filename` in interface `Icon`. Property `destination` only supports directories.

**Comment by Sideral-Inc** [comment]
@arthurbergmz

Same as @Jaid here. Also using 4.1.1 and there is no option to overwrite the default name (icon_${dimensions}).
